exporters/geneva/src/exporter.cc

```diff
@@ -0,0 +1,359 @@
+                       point_data_with_attributes.point_data)) {
+          auto value = nostd::get<sdk::metrics::HistogramPointData>(
+              point_data_with_attributes.point_data);
+          MetricsEventType event_type =
+              MetricsEventType::ExternallyAggregatedULongDistributionMetric;
+          if (nostd::holds_alternative<double>(value.sum_)) {
+            // TBD -- Not supported
+            continue;
+          }
+          body_length = SerializeHistogramMetrics(
+              sdk::metrics::AggregationType::kHistogram, event_type,
+              value.count_, value.sum_, value.min_, value.max_,
+              nostd::get<sdk::metrics::HistogramPointData>(
+                  point_data_with_attributes.point_data)
+                  .boundaries_,
+              nostd::get<sdk::metrics::HistogramPointData>(
+                  point_data_with_attributes.point_data)
+                  .counts_,
+              metric_data.end_ts, metric_data.instrument_descriptor.name_,
+              point_data_with_attributes.attributes);
+          data_transport_->Send(buffer_histogram_,
+                                body_length + kBinaryHeaderSize);
+        }
+      }
+    }
+  }
+  return opentelemetry::sdk::common::ExportResult::kSuccess;
+}
+
+bool Exporter::ForceFlush(std::chrono::microseconds timeout) noexcept {
+  return true;
+}
+
+bool Exporter::Shutdown(std::chrono::microseconds timeout) noexcept {
+  const std::lock_guard<opentelemetry::common::SpinLockMutex> locked(lock_);
+  is_shutdown_ = true;
+  return true;
+}
+
+size_t Exporter::InitializeBufferForNonHistogramData() {
+  // The buffer format is as follows:
+  // -- BinaryHeader
+  // -- MetricPayload
+  // -- Variable length content
+
+  // Leave enough space for the header and fixed payload
+  auto bufferIndex = kBinaryHeaderSize + kMetricPayloadSize;
+  SerializeString(buffer_non_histogram_, bufferIndex,
+                  connection_string_parser_.account_);
+  SerializeString(buffer_non_histogram_, bufferIndex,
+                  connection_string_parser_.namespace_);
+  return bufferIndex;
+}
+
+size_t Exporter::InitiaizeBufferForHistogramData() {
+  // The buffer format is as follows:
+  // -- BinaryHeader
+  // -- ExternalPayload
+  // -- Variable length content
+
+  // Leave enough space for the header and fixed payload
+  auto bufferIndex = kBinaryHeaderSize + kExternalPayloadSize;
+  SerializeString(buffer_histogram_, bufferIndex,
+                  connection_string_parser_.account_);
+  SerializeString(buffer_histogram_, bufferIndex,
+                  connection_string_parser_.namespace_);
+  return bufferIndex;
+}
+
+size_t Exporter::SerializeNonHistogramMetrics(
+    sdk::metrics::AggregationType agg_type, MetricsEventType event_type,
+    const sdk::metrics::ValueType &value, common::SystemTimestamp ts,
+    std::string metric_name, const sdk::metrics::PointAttributes &attributes) {
```

Comment:
Could `metric_name` be passed as const string referrence?

Reply:
Yes it should. Have modified it now.